http_server.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+import email.utils
+
+
+def server_socket_function():
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+
+            recieve_total = ""
+            buffersize = 32
+            finished = 0
+            while not finished:
+                recieve = conn.recv(buffersize)
+                if len(recieve) < buffersize:
+                    recieve_total += recieve
+                    finished = 1
+                else:
+                    recieve_total += recieve
+
+            if recieve_total:
+                try:
+                    response = parse_request(recieve_total)
+                except RequestError, msg:
+                    errors = str(msg).strip("'").split(' ', 1)
+                    response = response_error(errors[0], errors[1])
+                conn.sendall(response)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def response_ok():
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/plain'
+    body = '200 OK'
+    response_list = [first_line, timestamp, content_header, ' ', body, '\r\n']
```

Comment:
You've added a placeholder for the blank line, but it isn't blank.  It's a string with one space in it.